### PR TITLE
VC-4212 add account state

### PR DESCRIPTION
**Pull request by alex-bodnar**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/vipcoin/chain/accounts/accounts.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+		return pq.Int64Array{}, nil
+	}
+
+	query := `INSERT INTO vipcoin_chain_accounts_affiliates
+		 (address, affiliation_kind, extras)
+	 VALUES
+		 (:address, :affiliation_kind, :extras)
+	 RETURNING id`
+
+	resultID := make(pq.Int64Array, len(affiliates))
```
Comment (DirusK) [inline database/vipcoin/chain/accounts/accounts.go:85]:
```suggestion
	resultID := make(pq.Int64Array, 0, len(affiliates))
```

File: database/vipcoin/chain/accounts/accounts.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	if len(affiliatesID) == 0 {
+		return []*accountstypes.Affiliate{}, nil
+	}
+
+	arguments := map[string][]interface{}{"id": {}}
+	for _, val := range affiliatesID {
+		arguments["id"] = append(arguments["id"], val)
+	}
+
+	query, args := filter.NewFilter().SetArgumentsMap(arguments).
```
Comment (DirusK) [inline database/vipcoin/chain/accounts/accounts.go]: Можно сделать функцию которая распарсит айдишники в []interface{}, а дальше удобнее ложить в аргументы  
Пример:  
filter.SetArgument(fieldID,parserID.... )  
Но это такое, если понравится

File: database/vipcoin/chain/accounts/accounts.go (hunk 1 above)
Comment (alex-bodnar) [reply database/vipcoin/chain/accounts/accounts.go:85]: Мне нужен массив с нулевыми значениями, я в них на 93 строке записываю результат.

File: database/vipcoin/chain/accounts/accounts.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+	return accounts, nil
+}
+
+func getAffiliates(db *sqlx.DB, affiliatesID pq.Int64Array) ([]*accountstypes.Affiliate, error) {
+	if len(affiliatesID) == 0 {
+		return []*accountstypes.Affiliate{}, nil
+	}
+
+	query, args := filter.NewFilter().SetArgument("id", parsID(affiliatesID)...).
```
Comment (IamStubborN) [inline database/vipcoin/chain/accounts/accounts.go]:
```suggestion
	query, args := filter.NewFilter().SetArgument("id", parseID(affiliatesID)...).
```

**Review by IamStubborN — APPROVED** [review]
LGTM

**Comment by DirusK** [comment]
LGTM

**Pull request merged**